src/box/box.cc

```diff
@@ -782,28 +764,37 @@ box_check_election_timeout(void)
 	return d;
 }
 
-static void
-box_check_replication(void)
+static int
+box_check_uri_set_for_option(const char *option_name)
 {
-	int count = cfg_getarr_size("replication");
-	for (int i = 0; i < count; i++) {
-		const char *source = cfg_getarr_elem("replication", i);
-		if (box_check_uri(source, "replication") != 0)
-			diag_raise();
+	struct uri_set uri_set;
+	if (cfg_get_uri_set(option_name, &uri_set) != 0) {
+		uri_set_destroy(&uri_set);
+		return -1;
```

Comment:
You should convert the error returned by `cfg_get_uri_set` to `ClientError(ER_CFG)` here.

Reply:
fixed